src/network/networkd-network.c

```diff
@@ -1575,6 +1575,48 @@ int config_parse_section_route_table(
         return 0;
 }
 
+int config_parse_dhcp_max_client_attempt(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Network *network = data;
+        uint64_t k;
+        int r;
+
+        assert(network);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                network->dhcp_max_client_attempt = 0;
+                return 0;
+        }
+
+        if (streq(rvalue, "infinity"))
+                network->dhcp_max_client_attempt = (uint64_t) -1;
```

Comment:
Either use `MAX_CLIENT_ATTEMPT_INFINITY` here, or remove the `MAX_CLIENT_ATTEMPT_INFINITY` define.